### DISCUSSION TITLE
Cannot get authentication work

### DISCUSSION THREAD

**Discussion by tolgig** [strapi-community/plugin-io#71] [Q&A] [answered]
Hi,
I am trying to get the authentication to work. Unfortunately, so far without success.

The error message is:
"The function 'getAdvancedSettings' is not defined" is returned from the socket server.

The token is fine (it works flawlessly with REST).

When I don't use authentication, it works. As soon as the authentication is active, I get this error.

I remove the credentials:true but then it says there is no valid token. (but token is valid. Same token with rest is okay)

Frontend:

`    const JWT_TOKEN = getToken();

    if (!JWT_TOKEN) return;
    const socket = io("https://xxx234.com/", {
      transports: ["websocket"],
      auth: {
        strategy: "jwt",
        token: JWT_TOKEN,
      },
    });
    console.log("connecting");

    socket.on("connect", () => {
      console.log("connected");
      socket.on("chat-message:create", (data) => {
        console.log("chat-message created!");
        console.log(data);
      });
      socket.on("chat-messages:create", (data) => {
        console.log("chat-message created!");
        console.log(data);
      });
    });
    socket.on("disconnect", () => console.log("connected"));
    
    
    
    ---
    Here is my server config:

      io: {
    enabled: true,
    config: {
      contentTypes: ["api::chat-message.chat-message"],
      socket: {
        serverOptions: {
          cors: {
            origin: [
              "http://localhost:1337",
              "http://localhost:3000",
              "http://localhost:3001",
              "http://localhost:8080",
            ],
            methods: ["GET", "POST"],
            credentials: true,
          },
        },
      },
      events: [
		     {
		      name: 'connection',
				handler: ({ strapi }, socket) => {
					// will log every time a client connects
					strapi.log.info(`[io] a new client with id ${socket.id} has connected`);
				},
			},
		],
    },
  },

`

latest strapi version
latest strapi-plugin.io version

any ideas?

**Comment by ComfortablyCoding** [comment]
Sorry to hear that! This looks like a bug on the packages side. I'll have a look at it tonight.

**Reply by ComfortablyCoding** [reply, marked as answer]
This was a bug and should be fixed via #72 in the 2.0.4 release. Please try again with this new release/version and let me know if their is still any issue.

**Reply by tolgig** [reply]
it works. thanks again!

**Reply by ComfortablyCoding** [reply]
Your welcome.
